content/addons.md

```diff
@@ -138,20 +146,216 @@ There are several options to see the addon in action. We could use `npm link` or
+
+The best way to learn how to CSS preprocessors in your addon is to consult the documentation for the preprocessor addon of your choice, and study how other addon authors have implemented stylesheets. For example, [ember-styleguide](https://github.com/ember-learn/ember-styleguide/) is a UI component library that was made for the main Ember websites. It uses [ember-cli-sass](https://www.emberobserver.com/addons/ember-cli-sass) to manage styles. You can search [Ember Observer](https://emberobserver.com) for many more examples of styling in action!
+
+### Adding UI functionality with JavaScript
+
+There are two main types of JavaScript functionality that an addon can provide: API methods that developers can use after importing your addon, and interactivity JavaScript that is part of UI components. We'll cover UI use cases first.
+
+Interactivity in an addon can be handled the same way that it is done for an Ember app component. Every addon component template has a corresponding JavaScript file to go with it. For example, an addon can have its own actions, computed properties, and imported dependencies. Developers using the addon can pass in their own actions and attributes.
+
+For more information about building interactivity for your addon, reference the [Ember Guides components section](https://guides.emberjs.com/release/components/defining-a-component/). Remember to test as you work!
+
+### Providing multiple templates in one addon
+
 
-### Block form component template
+## Writing a JavaScript utilties addon
 
-### Adding JavaScript functionality
+Many addons have no UI components in them, or they offer a combination of JavaScript utilities and template helpers. In the regular npm ecosystem, JavaScript utility libraries are some of the most common packages. Although we could write a normal node package, providing an Ember addon to developers has some advantages. The developers don't need to worry about how to import a normal npm package. They can use `ember install our-addon-name` and get going right away.
+
+### Providing public API methods in the addon
+
+After we've created our addon file structure with `ember addon <addon-name>`, we can write some functions that will be available for an app to use. Such functions are commonly referred to as "public API." If the behavior of public API changes, it's convention in the Ember community to follow semver and change the major version of the addon. Semver is a cross-program-language versioning scheme that helps other developers or coworkers know which versions of a library will require them to refactor their apps. Learn more about semver [here](https://semver.org/).
+
+Similar to a normal npm package, the entry point is named `index.js`. The files exported from `addon/index.js` will be available to developers using the addon in their apps.
```

Comment:
The addon entry point, by default, is {addonName}/index.js (top level inside the addon). This is different from using {addonName}/addon/index.js  to export modules.